library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -269,6 +292,7 @@ public void flush() {
     repeatableControlSet = false;
     repeatableControlCc1 = 0;
     repeatableControlCc2 = 0;
+    incomingDataTargetChannel = NTSC_CC_CHANNEL_1;
```

Comment:
For data received prior to the first control character, both initially or after a flush, is assuming `incomingDataTargetChannel = NTSC_CC_CHANNEL_1` a possibly incorrect assumption? And if so would it be more correct to set to an `CHANNEL_UNKNOWN` constant or similar, in both cases, and drop all data until we see a control character to know what channel the data belongs to?

Reply:
You are right, CHANNEL_UNKNOWN might be technically more correct as we should not handle most bytes before identifying the context correctly. The side effect of this change could be that previously working captions for on-demand content not having correctly defined the channel at the very beginning will start missing the first few characters from now on.
 
Defaulting it to CHANNEL_1 might cause problems with live content at the very beginning, but most of the times - even for live content - only one caption channel is used, the decoder will automatically recover from this problem with the first byte pair containing the correct channel information.

As it seems live over-the-air use cases are relatively rare, (as no other people are creating pull requests :) ) it might be safer setting the defaults in a way that would not cause any regression even for not perfect 608 content. I have not tested this situation too much (probability of incorrect on-demand content Vs. probability of live content having more than one channels), so I do not have a definitive suggestion. Maybe CHANNEL_UNKNOWN, as this change is trying to introduce correct handling of channels.

Reply:
Thanks for the clarification. Let's leave it how it is for now. I'll add a comment explaining why as part of the merge. We can change it later if desired.